libminifi/test/unit/MemoryUsageTest.cpp

```diff
@@ -22,28 +22,28 @@
 
 #include <cstring>
 
+#include "utils/crypto/EncryptionUtils.h"
 #include "utils/gsl.h"
 #include "utils/OsUtils.h"
 #include "unit/TestBase.h"
 #include "unit/Catch.h"
 
 TEST_CASE("Test Physical memory usage", "[testphysicalmemoryusage]") {
   constexpr bool cout_enabled = true;
-  std::vector<char> v(30000000);
-  // use v to prevent recent compilers from optimizing out the allocation
-  strcpy(v.data(), " bytes\n");  // NOLINT: "almost always, snprintf is better than strcpy", but formatting is unnecessary here
+  const auto large_random_vector = minifi::utils::crypto::randomBytes(30'000'000);
```

Comment:
There's no need for crypto random here, we could use normal PRNG, so it works better on systems without a fast crypto random generator.

And it would be nice to add a comment about why it's random, like in the PR description.

Reply:
I have changed it to use `<random>` in fa35ebc33e8198fa11450183c9f54705c7b2073f.  I don't know if it works "better" (it is about 3 times slower on my computer, 0.3s instead of 0.1s), but I hope we'll have a few years until a compiler can optimize this out, too.

Reply:
Is the extra time spent in `std::accumulate`? We might get away with only accessing a couple of elements to make it faster.
Also, with this many elements, we're likely running into signed integer overflow, which is undefined behavior.

Reply:
`accumulate` is fast; calling `dist(gen)` is slow.  I have reduced the number of random numbers used from 30,000,000 to 40 in 4dddfd9b7e6563f0fe9c9f3ae2e2d5a073dd4161; now it runs in 0.02 seconds.